relay-system/src/service.rs

```diff
@@ -1046,12 +1047,12 @@ mod tests {
     impl Service for MockService {
         type Interface = MockMessage;
 
-        fn spawn_handler(self, mut rx: Receiver<Self::Interface>) {
+        fn spawn_handler(self, mut rx: Receiver<Self::Interface>) -> JoinHandle<()> {
```

Comment:
Note: by requiring that `spawn_handler` returns exactly one `JoinHandle`, we restrict the impl to define exactly one main task. Not sure if this is what we want, because the purpose of the spawn handler was to give the implementor more liberty. If we do restrict it, we might as well replace the trait method `spawn_handler` by a trait method `run`, and call `tokio::spawn` from the outside.